tests/dtypes_test.py

```diff
@@ -971,6 +985,17 @@ def testFloat8PromotionError(self):
                                   ".*8-bit floats do not support implicit promotion"):
         x + y
 
+  @jax.numpy_dtype_promotion('standard')
+  def testFloat4PromotionError(self):
+    for dtype in fp4_dtypes:
+      if dtype == dtypes.float4_e2m1fn and jtu.test_device_matches(['tpu']):
+        self.skipTest("TPU does not support float4_e2m1fn.")
+      x = jnp.array(1, dtype=dtype)
+      y = jnp.array(1, dtype='float32')
+      with self.assertRaisesRegex(dtypes.TypePromotionError,
+                                  ".*4-bit floats do not support implicit promotion"):
```

Comment:
No error matching this text is raised – you'll probably need to add a condition in `jax/_src/dtypes.py`.

Reply:
TypePromotionError is indeed thrown in a standalone test.